packages/react/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -177,6 +177,32 @@ export function createHooksInternal<
     return createTRPCClient(opts);
   };
 
+  /**
+   * To allow easy interactions with groups of related queries, such as
+   * invalidating all queries of a router, we use an array as the path when
+   * storing in tanstack query. This function converts from the `.` separated
+   * path passed around internally by both the legacy and proxy implementation.
+   * https://github.com/trpc/trpc/issues/2611
+   */
+  const getArrayQueryKey = (
+    queryKey: string | [string] | [string, ...unknown[]],
```

Comment:
We can only have 1 input arg

```suggestion
    queryKey: string | [string] | [string, unknown],
```

Reply:
Yeah ... I did think / try this too but this then doesn't play nice when the type of the `pathAndInput` passed to some of the implementations in the [`TRPCProvider` are `path: TPath, ...args: ProcedureArgs<inferProcedureParams<TProcedure>>`](https://github.com/trpc/trpc/blob/3741fc32d04ea8f6515e3df71234408170b4142a/packages/react/src/shared/hooks/createHooksInternal.tsx#L261) ... It won't happily take the `...args` type it has there and I can't allow that specific type as it is a generic ... 

Is this type actually wrong for the `pathAndInput` too? 

...This was my only escape hatch to allow the function to be used in all instances we are actually sneakily passing `queryKeys` in!